File: app/src/main/java/com/assignment/exceptions/FileAlreadyExistException.java
package com.assignment.exceptions;

public class FileAlreadyExistException extends BaseException {

    private static final String MESSAGE = "File already exits.";
    public FileAlreadyExistException() {

    }

    public FileAlreadyExistException(String message) {
        super(MESSAGE + " : "+ message);
    }

    public FileAlreadyExistException(String message, Throwable cause) {
        super(MESSAGE + " : " + message, cause);
    }
}

File: app/src/main/java/com/assignment/exceptions/FileAlreadyExistException.kt
package com.assignment.exceptions

class FileAlreadyExistException : BaseException {
    constructor() {

    }

    constructor(message: String) : super("$MESSAGE : $message") {}

    constructor(message: String, cause: Throwable) : super("$MESSAGE : $message", cause) {}

    companion object {

        private val MESSAGE = "File already exits."
    }
}
